crates/compiler/build/src/target.rs

```diff
@@ -76,7 +76,7 @@ pub fn arch_str(target: Target) -> &'static str {
         roc_target::Architecture::X86_32 if cfg!(feature = "target-x86") => "x86",
         roc_target::Architecture::Aarch64 if cfg!(feature = "target-aarch64") => "aarch64",
         roc_target::Architecture::Aarch32 if cfg!(feature = "target-arm") => "arm",
-        roc_target::Architecture::Wasm32 if cfg!(feature = "target-webassembly") => "wasm32",
+        roc_target::Architecture::Wasm32 if cfg!(feature = "target-wasm32") => "wasm32",
```

Comment:
Using the same helpers for llvm calls surfaced this issue, we were referencing a feature that doesn't exist.